flocker/acceptance/endtoend/test_leases.py

```diff
@@ -0,0 +1,229 @@
+        * Request a move or delete operation.
+        * Wait for a short time; enough time for an unexpected unmount to take
+          place.
+        * Restart the container and write data to it, to demonstrate that the
+          dataset is still mounted and writable.
+        * Stop the container again.
+        * Release the lease, allowing the previously requested operation to
+          proceed.
+        * Wait for the previously requested operation to complete.
+
+        :param Cluster cluster: The cluster on which to operate.
+        :param operation: The ``FlockerClient`` method to call before releasing
+            the lease.
+        :param dict additional_kwargs: Any additional arguments to pass to
+            ``operation``.
+        :param state_method: A callable which returns a ``Deferred`` that fires
+            when the requested operation has been performed.
+        :returns: A ``Deferred`` that fires when the all the steps have
+            completed.
+        """
+        http_port = 8080
+        dataset_id = uuid4()
+        datasets = []
+        leases = []
+        containers = []
+        client = get_docker_client(cluster, cluster.nodes[0].public_address)
+
+        creating_dataset = create_dataset(
+            self, cluster, maximum_size=REALISTIC_BLOCKDEVICE_SIZE,
+            dataset_id=dataset_id
+        )
+
+        def get_dataset_state(configured_dataset):
+            """
+            XXX: This shouldn't really be needed because ``create_dataset``
+            returns ``wait_for_dataset`` which returns the dataset state, but
+            unfortunately ``wait_for_dataset`` wipes out the dataset path for
+            comparison purposes.
+            """
+            d = cluster.client.list_datasets_state()
+            d.addCallback(
+                lambda dataset_states: [
+                    dataset_state
+                    for dataset_state in dataset_states
+                    if dataset_state.dataset_id == dataset_id
+                ][0]
+            )
+            d.addCallback(
+                lambda dataset: datasets.insert(0, dataset)
+            )
+            return d
+        getting_dataset_state = creating_dataset.addCallback(
+            get_dataset_state
+        )
+
+        def acquire_lease(ignored):
+            # Call the API to acquire a lease with the dataset ID.
+            d = cluster.client.acquire_lease(
+                dataset_id, UUID(cluster.nodes[0].uuid),
+                # Lease will never expire
+                expires=None
+            )
+            d.addCallback(lambda lease: leases.insert(0, lease))
+            return d
+        acquiring_lease = getting_dataset_state.addCallback(acquire_lease)
+
+        def start_http_container(ignored):
+            """
+            Launch data HTTP container and make POST requests
+            to it in a looping call every second.
+            return looping call deferred
+            """
```

Comment:
This docstring doesn't seem to match the code. 

It doesn't return a looping call deferred, and doesn't do any POSTing that I can see.